src/components/TermBox.vue

```diff
@@ -1,21 +1,40 @@
 <template>
-  <div class="termbox-placeholder">
-  </div>
+	<div class="termbox">
+		<div class="primary-language">{{primaryLanguageName}}</div>
+	</div>
 </template>
 
 <script lang="ts">
-import Vue from 'vue';
+import Vue, { VueConstructor } from 'vue';
 import Component from 'vue-class-component';
+import { instance as factory } from '../common/TermboxFactory';
+import { mapState } from 'vuex';
+import { NS_USER } from '@/store/namespaces';
+
+interface TermboxBindings extends Vue {
+
+	primaryLanguage: string;
+
+}
+
+@Component( {
+	computed: mapState( NS_USER, {
+		primaryLanguage: 'primaryLanguage',
+	} ),
+} )
+export default class TermBox extends ( Vue as VueConstructor<TermboxBindings> ) {
+	public primaryLanguageName = '';
+
+	public created() {
+		factory.getLanguageRepository()
+			.getLanguageName( this.primaryLanguage, this.primaryLanguage )
+			.then( ( languageName ) => {
```

Comment:
This will result in an update to `this.primaryLanguageName` _asynchronously_ which is not a problem in the client because resolution should be so quick that the visual effect can be disregarded, but creates a major problem in the server - as there is no way to wait for it this way.
I feel we may be better off with populating the store on dispatch and merely from it reading here.

Did some reading...
[0] https://ssr.vuejs.org/guide/data.html#data-store
[1] https://ssr.vuejs.org/guide/data.html#server-data-fetching
[2] https://ssr.vuejs.org/guide/routing.html#routing-with-vue-router
[3] https://github.com/iamfunsho/vue-ssr-example/blob/master/src/entry-server.js
[4] https://vuejsdevelopers.com/2017/12/11/vue-ssr-router/
[5] https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards